### DISCUSSION TITLE
How to test the throughput of Tendermint or my Application?

### DISCUSSION THREAD

**Discussion by ruojichong** [tendermint/tendermint#9066] [General]
I notice that the tendermint or my application can only process 25tx/s, is this normal？ Can tendermint process transactions over 100tx/s? pls, help.
my tendermint version is 0.34.19 and i only use one node in local network.

**Comment by creachadair** [comment]
Hi, @ruojichong. There is no simple, universal answer. To predict the performance of a consensus network from tests run on a single node. Consensus throughput depends on the size and connectivity of the network, among other things. It will also depend on how you've set up your load test (e.g., how transactions are being added, and at what rate, settings for block and transaction size, mempool size, what your application is doing, etc.).

A good place to start would be to diagnose where your single-node test network is spending its time (e.g., executing transactions, waiting for input, storing data, etc.). I'd suggest you begin by enabling pprof (`rpc.pprof_laddr`) and capturing some CPU and memory traces from the running node under your load test. If you're new to pprof, I'd suggest reading through [Profiling Go Programs](https://go.dev/blog/pprof) as a way to get oriented.
